tests/functional/docs/test_shared_example_config.py

```diff
@@ -0,0 +1,124 @@
+import botocore.session
+from botocore.model import OperationNotFoundError
+from botocore.utils import parse_timestamp
+
+
+def test_lint_shared_example_configs():
+    session = botocore.session.Session()
+    loader = session.get_component('data_loader')
+    services = loader.list_available_services('examples-1')
+    for service in services:
+        service_model = session.get_service_model(service)
+        example_config = loader.load_service_model(
+            service, 'examples-1', service_model.api_version
+        )
+        examples = example_config.get("examples", {})
+        for operation, operation_examples in examples.items():
+            for example in operation_examples:
+                yield _lint_single_example, operation, example, service_model
+
+
+def _lint_single_example(operation_name, example_config, service_model):
+    # The operation should actually exist
+    assert_operation_exists(service_model, operation_name)
+    operation_model = service_model.operation_model(operation_name)
+    assert_valid_values(service_model.service_name, operation_model,
+                        example_config)
+
+
+def assert_valid_values(service_name, operation_model, example_config):
+    example_input = example_config.get('input')
+    input_shape = operation_model.input_shape
+    example_id = example_config['id']
+
+    if input_shape is None and example_input:
+        raise AssertionError(
+            "Input found in example for %s from %s with id %s, but no input "
+            "shape is defined." % (
+                operation_model.name, service_name, example_id
+            ))
+
+    example_output = example_config.get('output')
+    output_shape = operation_model.output_shape
+
+    if output_shape is None and example_output:
+        raise AssertionError(
+            "Output found in example for %s from %s with id %s, but no output "
+            "shape is defined." % (
+                operation_model.name, service_name, example_id
+            ))
+
+    try:
+        if example_input is not None and input_shape is not None:
+            _assert_valid_values(input_shape, example_input)
+
+        if example_output is not None and output_shape is not None:
+            _assert_valid_values(output_shape, example_output)
+    except AssertionError as e:
+        raise AssertionError(
+            "Invalid value in example for %s from %s with id %s: %s" % (
+                operation_model.name, service_name, example_id, e
+            ))
+
+
+def _assert_valid_values(shape, example_value, path=None):
+    if path is None:
+        path = []
+    path.append(shape.name)
+
+    if shape.type_name == 'timestamp':
+        _assert_valid_timestamp(example_value, path)
+    elif shape.type_name == 'structure':
+        _assert_valid_structure_values(shape, example_value, path)
+    elif shape.type_name == 'list':
+        _assert_valid_list_values(shape, example_value, path)
+
```

Comment:
You probably want to handle maps as well because you can recurse through those as well.